Task 40

eisina opened the pull request:
WebDriver manager class updated and implicit waiter added.
SeleniumEasyTest class created with the following tests:
- test with multiselect option
- tests for alerts (confirming, cancelling and sending values to alert box)
- automated script, which waits for user to appear
- automated script for refresh page when percentage is >= 50
- automated script for returning list of custom objects (class fields: Name, Position, Office) with age > x and salary <= y for all pages

eisina committed:
Yandex login test is parametrized, added explicit and implicit waiters.
WebDriver manager class updated and implicit waiter added.
SeleniumEasyTest class created with the following tests:
-test with multiselect option
-tests for alerts (confirming, cancelling and sending values to alert box)
- automated script, which waits for user to appear
- automated script for refresh page when percentage is >= 50
- automated script for returning list of custom objects (class fields: Name, Position, Office) with age > x and salary <= y for all pages

KBugrimCS commented on src/test/coherentsolutions.web.test/java/Managers/WebDriverManager.java:
```diff
@@ -3,17 +3,20 @@
     public static WebDriver initDriver() {
         driver = new ChromeDriver();
         driver.manage().window().maximize();
+        driver.manage().timeouts().implicitlyWait(3, TimeUnit.SECONDS);
```
3 seconds is very short timeout, i suggest to increase this value to 10 or even 15 sec

KBugrimCS commented on src/test/coherentsolutions.web.test/java/YandexTest.java:
```diff
@@ -5,29 +5,39 @@
-    public void loginTest() throws InterruptedException, IOException {
+    @Test(description = "Checking login with valid credentials", dataProvider = "credentials")
+    public void loginTest(String login, String password) throws InterruptedException {
         driver.get("https://mail.yandex.com/");
```
Please move the site url to Environment properties

KBugrimCS commented on src/test/coherentsolutions.web.test/java/YandexTest.java:
```diff
@@ -36,18 +46,32 @@ public void loginTest() throws InterruptedException, IOException {
+                "Login Page is opened, user is logged out.");
+        log.info("User logged out successfully");
+
+        driver.manage().deleteAllCookies();
```
move this logic to after method

KBugrimCS commented on src/test/coherentsolutions.web.test/java/SeleniumEasyTest.java:
```diff
@@ -0,0 +1,196 @@
+
+    @Test(description = "Checking selecting multiselect option")
+    public void multiselectTest() {
+        driver.get("https://demo.seleniumeasy.com/basic-select-dropdown-demo.html");
```
Please move the site url to Environment properties

KBugrimCS commented on src/test/coherentsolutions.web.test/java/SeleniumEasyTest.java:
```diff
@@ -0,0 +1,196 @@
+
+    @Test(description = "Checking acceptingJava Script Confirm Box")
+    public void confirmBoxAcceptTest() {
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
```
the same

KBugrimCS commented on src/test/coherentsolutions.web.test/java/SeleniumEasyTest.java:
```diff
@@ -0,0 +1,196 @@
+
+    @Test(description = "Checking cancelling Java Script Confirm Box")
+    public void confirmBoxCancelTest() {
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
```
the same

KBugrimCS commented on src/test/coherentsolutions.web.test/java/SeleniumEasyTest.java:
```diff
@@ -0,0 +1,196 @@
+    public void waitForUserTest() {
+        driver.get("https://demo.seleniumeasy.com/dynamic-data-loading-demo.html");
+        driver.findElement(Locators.GET_NEW_USER_BUTTON).click();
+        webDriverWait.pollingEvery(Duration.ofSeconds(3)).until(ExpectedConditions.elementToBeClickable(Locators.USER_IMAGE));
```
very huge polling interval

KBugrimCS commented on src/test/coherentsolutions.web.test/java/SeleniumEasyTest.java:
```diff
@@ -0,0 +1,196 @@
+
+    @Test(description = "Checking refreshing the page when download percentage >50")
+    public void refreshDownloadTest() {
+        driver.get("https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html");
```
the same

KBugrimCS commented on src/test/coherentsolutions.web.test/java/SeleniumEasyTest.java:
```diff
@@ -0,0 +1,196 @@
+        driver.get("https://demo.seleniumeasy.com/table-sort-search-demo.html");
+        Select elementNumber = new Select(driver.findElement(Locators.SHOW_ENTRIES_DROPDOWN));
+        elementNumber.selectByValue("10");
+        log.info("Option '10' is selected in dropdown “Show () entries”");
```
need to add assertion here that '10' value is selected

eisina committed:
Fixing comments:
- Class ConstantUtils added with urls
- All tests modified to use urls from constants
- Implicit wait timeout changed
- After method is added to Yandex test
- Changing polling interval
- Assertion added to returnEmployeeTest
